### PR TITLE
contribution

### PR DESCRIPTION
**Pull request by BilalAlam173**

### CODE REVIEW

**Comment by mashhoodr** [comment]
@afirdousi available for your review. This just has static login and signup pages we will add Firebase and link them up in the next step. Since these pages are isolated, they wont effect rest of the website at the moment.

**Comment by mashhoodr** [comment]
@BilalAlam173 why did you remove the fsevents from package.json?

**Comment by BilalAlam173** [comment]
@mashhoodr  fsevent was automaticaly removed in package-lock.json.I will add it again and create another pull request.

**Comment by mashhoodr** [comment]
oh its package-lock - @afirdousi was this something you used specifically on your end only? Since its not in package.json

**Comment by BilalAlam173** [comment]
@mashhoodr I have connected both signup and login pages with the database, both are working, I have not done any work on the callback of login as per your instruction, successful login only logs the user detail in the console.

**Comment by mashhoodr** [comment]
@afirdousi awaiting review.

File: server.js [hunk 1]
```diff
@@ -8,19 +8,32 @@ const db = require('./server/db/db');
...
 const app = express();
 
+
 // Parsers for POST data
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 
+//to allow Cross origin requests
+app.use(function(req, res, next) {
+  res.setHeader("Access-Control-Allow-Origin", "*");
```
Comment (mohuk) [inline server.js]: `"Access-Control-Allow-Origin", "*"` don't allow this. This header shouldn't be there in the first place, use some kind of reverse proxy to mitigate `CORS`

File: src/app/login/login.component.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<section id="testimonial" class="section-padding">
+    <div class="container">
+        <div class="buffer">
+        </div>
+        <div class="row">
+            <div class="col-md-4">
+            </div>
+            <div class="col-md-4">
+                <form role="form" action="javascript:;" method="post" id="login-form" autocomplete="off">
```
Comment (mohuk) [inline src/app/login/login.component.html]: `action="javascript:;" method="post"`

I believe we need to convert this into an `ngForm`

File: src/app/services/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    return this.http.post(this.api+'/login', body, options)
+                .map(res => res.json())
+                .catch(this.handleError);
+
+  }
+  
+
+  private handleError (error: Response | any) {
```
Comment (mohuk) [inline src/app/services/user.service.ts]: Error handling can be improved. `handleError` method will need to be copied where ever required. We can move this into central error handlers or interceptors.

File: src/app/signup/signup.component.html [hunk 4]
```diff
@@ -0,0 +1,96 @@
+<section id="testimonial" class="section-padding">
+    <div class="container">
+        <div class="buffer">
+        </div>
+        <div class="row">
+            <form role="form" action="javascript:;" method="post" id="login-form" autocomplete="off">
```
Comment (mohuk) [inline src/app/signup/signup.component.html]: same form issues :(

File: src/app/signup/signup.component.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  ngOnInit() {
+    console.log(this.User);
+  }
+
+  signup(){
+    this.UserService.save(this.User).subscribe( data => {
+      console.log(data);
+    });
+  }
+
```
Comment (mohuk) [inline src/app/signup/signup.component.ts]: :unamused: whitespace

**Comment by mohuk** [comment]
@mashhoodr If we are thinking to move to Firebase we might not need all this. We can simply use available authentication options in Firebase

File: src/app/login/login.component.html (hunk 2 above)
Comment (BilalAlam173) [reply src/app/login/login.component.html]: sure i will, in my next commit.

File: server.js (hunk 1 above)
Comment (BilalAlam173) [reply server.js]: https://www.npmjs.com/package/cors-anywhere can I use this as an alternate?

Comment (mohuk) [reply server.js]: You might not need anything such as this package

Comment (mohuk) [reply server.js]: `CORS` is not required. For development environment use proxying present in `angular-cli`. We can discuss the deployment with @afirdousi

Comment (BilalAlam173) [reply server.js]: I understand now. Thanks

Comment (afirdousi) [reply server.js]: I agree to @mohuk. I don't think we would need CORS.

**Comment by afirdousi** [comment]
Thanks a bunch for the amazing effort @BilalAlam173. I think this is a good first step moving forward.

My comment ( @mashhoodr @mohuk ) :

- I think we can start with this and later move to `firebase`? I understand firebase is super easy to configure but I am sure **more** developers have used simple `mongoose` with `mongo` on node than firebase (at the moment - maybe I am wrong about devs there ). And if my assumption is correct, we can start with this so more folks can contribute. We will sure have a req open for next quarter to move stuff to firebase. I know it will be redundant effort but may be more learning for new folks who are looking to contribute? 

Sounds good? If @mohuk & @mashhoodr agree, @BilalAlam173 we will need to add a simple profile page and redirect the user to it after both successful sign-in and sign up. I can provide you a simple sketch for a super simple profile page. It will have basic user info + a list of meetups user has signed up for. This meetup portion can be added later after we work on meetup resources section.

**Comment by mashhoodr** [comment]
@afirdousi yes, even though this way we have a greater overhead on deployment, db migrations and management - this should do it for now. 

@afirdousi profile page will be a new issue. You can create and add details for that.

The core thing remaining in this will be setting the session on the client side once this is done we can merge this and close the ticket.

**Comment by afirdousi** [comment]
@BilalAlam173 can you work on user session in coordination with @mashhoodr and we will be good to go. I will merge the PR.

**Comment by afirdousi** [comment]
This PR addresses [Issue#7](https://github.com/angular-pakistan/ngpakistan.com/issues/7)

**Comment by BilalAlam173** [comment]
@afirdousi thanks, yes I will start working on user session once I get those issue fixed pointed out by @mohuk in my next commit. @mashhoodr  can give me the instructions on user sessions, I will start working on it next.

File: src/app/services/error.service.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class ErrorService {
+
+  constructor() { }
+
+   handleError (error: Response | any) {
+    // In a real world app, we might use a remote logging infrastructure
```
Comment (afirdousi) [inline src/app/services/error.service.ts:9]: Well, its almost a real-world app isn't it 😏 ? jk ... we should definitely log errors remotely in the future.

**Comment by afirdousi** [comment]
Since we do not have auto-deployment / CI setup, ill setup it manually soon.

**Pull request merged**